bee-hive/bee_hive/run_workflow.py

```diff
@@ -25,19 +31,129 @@ def load_agent(agent):
         agent_store = json.load(f)
     return agent_store[agent]
 
-
+# --- Abstract Base Class for Agent Runners ---
+class AgentRunner(ABC):
+    def __init__(self, agent_name):
+        self.agent_name = agent_name
+
+    @abstractmethod
+    def run(self, prompt):
+        """Runs the agent with the given prompt."""
+        pass
+    
+    @abstractmethod
+    def stream(self, prompt):
+        """Runs the agent in streaming mode with the given prompt."""
```

Comment:
Add some comments to at least the interface methods `run` and `stream`

Reply:
We need comments in the whole module. Actually I think we need to split the file soon which will improve clarity & reduce edit conflicts
I plan to add for the additional code I added